### ISSUE TITLE
Memory error for batch size greater than 3

### ISSUE THREAD

**Issue by hamzarawal** [nianticlabs/monodepth2#236]
Hi, when I run the code for batch size greater than 3 I get CUDA out of memory error although memory is available. Here is the error stack. My PyTorch version is 1.6.0

Exception has occurred: RuntimeError
CUDA out of memory. Tried to allocate 20.00 MiB (GPU 0; 7.93 GiB total capacity; 2.81 GiB already allocated; 25.38 MiB free; 3.06 GiB reserved in total by PyTorch)
  File "/media/cvlab/cvlab1/hamza/ETRI/codes/monodepth2/layers.py", line 248, in forward
    SSIM_d = (mu_x ** 2 + mu_y ** 2 + self.C1) * (sigma_x + sigma_y + self.C2)
  File "/media/cvlab/cvlab1/hamza/ETRI/codes/monodepth2/trainer.py", line 412, in compute_reprojection_loss
    ssim_loss = self.ssim(pred, target).mean(1, True)
  File "/media/cvlab/cvlab1/hamza/ETRI/codes/monodepth2/trainer.py", line 438, in compute_losses
    reprojection_losses.append(self.compute_reprojection_loss(pred, target))
  File "/media/cvlab/cvlab1/hamza/ETRI/codes/monodepth2/trainer.py", line 265, in process_batch
    losses = self.compute_losses(inputs, outputs)
  File "/media/cvlab/cvlab1/hamza/ETRI/codes/monodepth2/trainer.py", line 210, in run_epoch
    outputs, losses = self.process_batch(inputs)
  File "/media/cvlab/cvlab1/hamza/ETRI/codes/monodepth2/trainer.py", line 194, in train
    self.run_epoch()
  File "/media/cvlab/cvlab1/hamza/ETRI/codes/monodepth2/train.py", line 18, in <module>
    trainer.train()

**Comment by mrharicot** [comment]
Hi,
What parameters are you running the training sessions with? Did you modify anything?
Can you run `nvidia-smi` and see how much free memory you have before running the training session?

**Comment by hamzarawal** [comment]
Hi, 
Thank you so much. I had previously checked `nvidia-smi` but did not find any process listed using the GPU. However now when I looked closely, although the processes were not listed, the memory was still occupied. I had to manually search for the processes using `sudo fuser -v /dev/nvidia*`. This lists the processes that have NVIDIA GPU device nodes open. 
Then I manually killed the processes using `sudo kill <pid>`. Memory is available now and higher batch sizes are working.
